x/interchainqueries/keeper/msg_server.go

```diff
@@ -82,12 +89,15 @@ func (k msgServer) RemoveInterchainQuery(goCtx context.Context, msg *types.MsgRe
 		return nil, sdkerrors.Wrapf(err, "failed to get query by query id: %v", err)
 	}
 
-	if query.GetOwner() != msg.GetSender() {
+	timeoutBlock := query.LastSubmittedResultLocalHeight + k.GetParams(ctx).QuerySubmitTimeout
```

Comment:
what you guys think about saving the QuerySubmitTimeout as well as we save QueryDeposit individually for each query? e.g. I registered a query when the module's QuerySubmitTimeout was of 50 blocks hardly relying on this value, and later if this value changes to 30, I will 1) lose money; 2) lose query.

Reply:
Good catch! Yes, I guess we should store `QuerySubmitTimeout` into `RegisteredQuery` as well. Otherwise it may lead to different confustions.